### PR TITLE
pkg install honors versions when checking for conflicts

### PR DESCRIPTION
**Pull request by shomron**
This commit allows installing a package when a different version is
already installed. Currently, if the package is installed globally (the
only option) - it will overwrite the previous version reference.
A followup change will allow installing into environments to work around
this limitation.

Part of #660

Also:

* Add PackageManager.IsInstalled - fuzzy-match installation check for packages, without parsing their manifests

Signed-off-by: Oren Shomron <shomron@gmail.com>

### CODE REVIEW

**Comment by shomron** [comment]
Exercising the change:

```
ks init multi && cd multi

ks --verbose=5 pkg install incubator/mysql
grep -A4 libraries app.yaml
tree vendor

ks --verbose=5 registry set incubator --uri https://github.com/shomron/parts/tree/nested-registry/nested/registry/incubator

# Overwrites global reference, installs new version
ks --verbose=5 pkg install incubator/mysql
grep -A4 libraries app.yaml
tree vendor

# Same version, show conflict
ks --verbose=5 pkg install incubator/mysql
```

File: pkg/registry/cache_test.go [hunk 1]
```diff
@@ -59,9 +67,16 @@ func Test_CacheDependency(t *testing.T) {
 		}
 		for _, lib := range libs {
 			a.On("UpdateLib", lib.Name, lib).Return(nil)
+
+			var checker installedChecker
+			// checker.On("IsInstalled", pkg.Descriptor{
```
Comment (bryanl) [inline pkg/registry/cache_test.go]: Is this coming back in the future, or should it be deleted?

**Comment by bryanl** [comment]
Will approve after the unused code is deleted.

File: pkg/registry/cache_test.go (hunk 1 above)
Comment (shomron) [reply pkg/registry/cache_test.go]: The test may become more elaborate in the future - for now I'll take these out. 👍

**Comment by shomron** [comment]
Removed unused code.

**Pull request merged**